File: cpp/07.cpp
#include <iostream>
#include <vector>
#include <chrono>

int main() {
    // Define the size of memory-intensive data structure
    const int SIZE = 1000000; // Adjust size as needed

    // Start the timer
    auto start = std::chrono::high_resolution_clock::now();

    // Allocate memory-intensive data structure (example: vector of integers)
    std::vector<int> data(SIZE);

    // Perform memory-intensive operations (example: fill the vector with values)
    for (int i = 0; i < SIZE; ++i) {
        data[i] = i;
    }

    // Stop the timer
    auto end = std::chrono::high_resolution_clock::now();
    std::chrono::duration<double> duration = end - start;

    // Display the time taken and memory usage
    std::cout << "Time taken: " << duration.count() << " seconds" << std::endl;
    std::cout << "Memory used: " << SIZE * sizeof(int) / (1024 * 1024) << " MB" << std::endl;

    return 0;
}

File: python/07.py
import time
import sys

def main():
    # Define the size of memory-intensive data structure
    SIZE = 1000000  # Adjust size as needed

    # Start the timer
    start = time.time()

    # Allocate memory-intensive data structure (example: list of integers)
    data = [0] * SIZE

    # Perform memory-intensive operations (example: fill the list with values)
    for i in range(SIZE):
        data[i] = i

    # Stop the timer
    end = time.time()
    duration = end - start

    # Display the time taken and memory usage
    print("Time taken:", duration, "seconds")
    print("Memory used:", sys.getsizeof(data) / (1024 * 1024), "MB")

if __name__ == "__main__":
    main()
